### PR TITLE
Add additional content sections from components page wireframe

### PR DESCRIPTION
**Pull request by cfarm**
Fleshing out the components page with all the content elements from the wireframe, minus Jinja templates for now (see Todos below for why)

## Additions

- specifications field which uses a Markdown/rich text editor widget, I've tested it out with a markdown table built at https://www.tablesgenerator.com/markdown_tables over here https://github.com/test-things/design-system/pull/21
  - preview the specifications table at: https://deploy-preview-21--test-design-system.netlify.com/design-system/components/atomic-components yes there are typos
- contributors placeholder avatars
- show code & specs button - this doesn't do anything yet, but a future PR will add the toggle JS.
- "Related items" and "Help us make improvements" links
- new template class name for all page templates in the CMS: uses atomic naming, for this page it's `t-component`. this class name is dynamic based on the `layout` field in the yaml front matter. for other pages it would be different:
  - `layout: component` means the class name is `t-component` on component pages that use the component page template: https://raw.githubusercontent.com/test-things/design-system/master/docs/components/bar-charts.md
  - `layout: homepage` means the class name is `t-homepage` on pages that use the homepage template: https://raw.githubusercontent.com/test-things/design-system/master/docs/pages/home.md

## Removals

-

## Changes

- lots of HTML with layout classes to mimic the layout of the wireframe
- 

## Testing

1.

## Notes

-

## Todos

- [ ] TENTATIVE  - Add Jinja code snippet tab in another PR. I'm planning to rename the existing code snippet widget to differentiate the two (HTML vs Jinja) and this will be easier in a smaller separate PR. - will discuss best way forward with chris first. We won't invest much time in this during sprint 2, as sprint 3 is when we'll focus on CMS fields. and I'll want the whole team's input on how to handle Jinja templates at that stage
- [ ] Toggle JS for show code & specs button

## Checklist

- [ ] PR has an informative and human-readable title
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code can be automatically merged (no conflicts)
- [ ] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [ ] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future todos are captured in comments
- [ ] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [ ] Project documentation has been updated
- [ ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### Browsers

- [ ] Chrome on desktop
- [ ] Firefox
- [ ] Safari on macOS
- [ ] Edge
- [ ] Internet Explorer 9, 10, and 11
- [ ] Safari on iOS
- [ ] Chrome on Android

### Accessibility

- [ ] Keyboard friendly
- [ ] Screen reader friendly

### Other

- [ ] Is useable without CSS
- [ ] Is useable without JS
- [ ] Flexible from small to large screens
- [ ] No linting errors or warnings
- [ ] JavaScript tests are passing

### CODE REVIEW

File: docs/_layouts/component.html [hunk 1]
```diff
@@ -9,29 +9,45 @@
...
+                                a-btn_icon__on-right">{% include icons/edit.svg %}</span>
+                    </a>
+                </div>
+            </div>
+        </div><!-- .content-l -->
 
         {% capture intro %}{{ page.intro | markdownify }}{% endcapture %}
         {% include intro.html %}
 
         {% for variation in page.variations %}
```
Comment (marteki) [inline docs/_layouts/component.html]: Thinking: how could we make sure that there's always at least one default state for a component, with variations being built on top of it?

File: docs/_layouts/component.html [hunk 2]
```diff
@@ -9,29 +9,45 @@
...
 
-        <h2>{{ page.title }}</h2>
+                    {% include status.html %}
 
-        {% include status.html %}
+                    {% include contributors.html %}
+                </div>
+            </div>
+            <div class="content-l_col content-l_col-1-3">
+                <div class="block">
```
[the remark this answers is not in the document]
Comment (marteki) [reply docs/_layouts/component.html]: Discussed internally

**Comment by cfarm** [comment]
@designlanguage can you review this example page for all the content types that are in the wireframe? only thing that should be missing is the jinja code snippet tab.

https://deploy-preview-21--test-design-system.netlify.com/design-system/components/atomic-components

**Comment by contolini** [comment]
This is dope @cfarm. How do you envision contributors being populated? Leave placeholders in the template and update it via AJAX on page load?

**Comment by cfarm** [comment]
@contolini yep, we might have to have a field for linking to the github component to use in the ajax request. that might be the hard part, figuring out what files count for tracking contributions. although, this reminds me, we might want to link to the codebase for a particular component in the documentation, which isn't part of the page content currently, so I'd recommend we add it.

**Comment by cfarm** [comment]
2 green checks, oh what a delight 😊

**Pull request merged**